lua/dressing/input.lua

```diff
@@ -91,6 +104,31 @@ M.history_next = function()
   end
 end
 
+---@param mode dressing.Mode?
+M.set_mode = function(mode)
+  if mode == "normal" then
+    vim.cmd("stopinsert")
+  elseif mode == "insert" then
+    vim.cmd("startinsert!")
+  elseif mode == "visual" then
+    vim.api.nvim_command("normal! vg_")
+  elseif mode == "select" then
+    M.set_mode("visual")
+    vim.api.nvim_feedkeys(vim.api.nvim_replace_termcodes("<C-g>", true, false, true), "n", true)
+  end
+end
+
+---@param mode dressing.Mode?
+M.restore_mode = function(mode)
+  if mode == "normal" then
```

Comment:
I think we should do
```suggestion
  if mode ~= "insert" then
```
I think the default should be going back to normal mode, and only restore insert if we started in insert

Reply:
In this case, nil also stops insert. Is that desired?

Reply:
Yeah, I think that we have a good idea how to restore insert mode (you just go back to insert mode), but any other mode not so much. I think that it makes more sense for any given mode to be returned to normal mode than to insert mode. For example, if I started in visual mode I would be surprised if after a `vim.ui.input` I found myself in insert mode.